### PR TITLE
Feature: Add support for directory getOrganizations() and getOrganizationMappings()

### LINKED ISSUE #313: Add organization support / organization mapping support for directories
It seems we are allowed to get the directories for an organization, but not the other way around when we have an account -> directory(s) -> organization(s).  This would be helpful in cases where users login via the main domain rather than a tenant sub domain and we do an automatic redirect.

### PR DESCRIPTION
**Pull request by typerandom**
Adds ability to retrieve organization and organization mappings from a directory.

Fixes #313.

### CODE REVIEW

**Comment by rdegges** [comment]
+1 good to merge

File: test/sp.resource.directory_test.js [hunk 1]
```diff
@@ -326,7 +328,83 @@ describe('Resources: ', function () {
...
+
+          directory.getOrganizationMappings(cbSpy);
+          directory.getOrganizationMappings(opt, cbSpy);
+        });
+
+        after(function () {
+          sandbox.restore();
+        });
+
+        it('should get organization mappingss', function () {
```
Comment (timothyej) [inline test/sp.resource.directory_test.js]: Typo: `mappingss` => `mappings`

**Comment by timothyej** [comment]
I've verified that the functionality of this PR works as advertised, and that the tests run successfully. Please fix the typo and then merge :+1: :)

**Comment by timothyej** [comment]
I fixed the typo myself, merging now :+1:

**Pull request merged**